### ISSUE TITLE
Android: cannot launch workpress example - can launch in Calabash Android 0.5.13.pre1

### ISSUE THREAD

**Issue by jmoody** [calabash/calabash#195]
### Reproduce

1. One Android device connected and on the same network.
2. Pulled the latest from develop.

```
$ cd samples/wordpress
$ bundle
$ be calabash build features/prebuilt/wordpress_android.apk
$ CAL_DEBUG=1 be cucumber -p android
```

### Expected

The app to launch.

### Found

```
Ensuring /Users/moody/git/calabash/calabash/samples/wordpress/test_servers/4eeff36cdcf5fb8ff1847c0559df688c_1.9.9.pre2.apk is installed
ADB Command: -s, 015d168974380006, shell
ADB input: ["pm list packages; echo \"__CAL_END__$?\"; exit 0"]
Application is already installed. Ensuring right checksum
ADB Command: -s, 015d168974380006, shell
ADB input: ["pm list packages -f; echo \"__CAL_END__$?\"; exit 0"]
ADB Command: -s, 015d168974380006, shell
ADB input: ["md5 '/data/app/org.wordpress.android.test-1.apk'; echo \"__CAL_END__$?\"; exit 0"]
ADB Command: -s, 015d168974380006, shell
ADB input: ["pm list packages; echo \"__CAL_END__$?\"; exit 0"]
ADB Command: -s, 015d168974380006, shell
ADB input: ["am start -e method clear -n org.wordpress.android.test/sh.calaba.instrumentationbackend.StatusReporterActivity; echo \"__CAL_END__$?\"; exit 0"]
ADB Command: -s, 015d168974380006, shell
ADB input: ["pm list packages; echo \"__CAL_END__$?\"; exit 0"]
Starting 'sh.calaba.instrumentationbackend.ClearAppData2' using: 'am instrument  org.wordpress.android.test/sh.calaba.instrumentationbackend.ClearAppData2'
ADB Command: -s, 015d168974380006, shell
ADB input: ["am instrument  org.wordpress.android.test/sh.calaba.instrumentationbackend.ClearAppData2; echo \"__CAL_END__$?\"; exit 0"]
Adb shell command exited with 1
Error message from ADB: 
android.util.AndroidException: INSTRUMENTATION_FAILED: org.wordpress.android.test/sh.calaba.instrumentationbackend.ClearAppData2
        at com.android.commands.am.Am.runInstrument(Am.java:865)
        at com.android.commands.am.Am.onRun(Am.java:282)
        at com.android.internal.os.BaseCommand.run(BaseCommand.java:47)
        at com.android.commands.am.Am.main(Am.java:76)
        at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
        at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:243)
        at dalvik.system.NativeStart.main(Native Method)

  Failed to clear app data: Adb shell command exited with 1: android.util.AndroidException: INSTRUMENTATION_FAILED: org.wordpress.android.test/sh.calaba.instr... (RuntimeError)
  /Users/moody/git/calabash/calabash/lib/calabash/android/device.rb:591:in `rescue in ts_clear_app_data'
  /Users/moody/git/calabash/calabash/lib/calabash/android/device.rb:588:in `ts_clear_app_data'
  /Users/moody/git/calabash/calabash/lib/calabash/android/device.rb:639:in `_clear_app_data'
  /Users/moody/git/calabash/calabash/lib/calabash/device.rb:78:in `clear_app_data'
  /Users/moody/git/calabash/calabash/lib/calabash/life_cycle.rb:91:in `clear_app_data'
  /Users/moody/git/calabash/calabash/samples/wordpress/features/support/hooks.rb:86:in `reset'
  /Users/moody/git/calabash/calabash/samples/wordpress/features/support/hooks.rb:53:in `on_new_scenario'
  /Users/moody/git/calabash/calabash/samples/wordpress/features/support/hooks.rb:8:in `Before'
```

### Notes

The Wordpress app did install.

Trying to launch from the console, results in an indefinite hang:

```
$ be calabash console features/prebuilt/wordpress_android.apk 
```

I also could not launch the spec/resources/android/xtc-android-sample.apk.

I am able to launch both apps with 0.5.13.pre1

**Comment by jmoody** [comment]
iOS app launches out of the box.

**Comment by jmoody** [comment]
Added some of the debug output to get the adb stack trace.

Wow!  That's a lot of logging!

**Comment by TobiasRoikjer** [comment]
@jmoody Could you please run this while running `adb logcat`? It should have a stacktrace

**Comment by jmoody** [comment]
Fixed by #208
